### ISSUE TITLE
Buffer menu sort order?

### ISSUE THREAD

**Issue by parkerault** [onivim/oni#2260]
Is there an option to change the sort order of the buffer menu (<C-tab>)? Most buffer menu plugins sort the buffer list in most recently used order, but the oni buffer menu sorts them by buffer number, which is a lot less useful. Is there a configuration setting for this or is the sort order hard coded?

**Comment by Akin909** [comment]
@parkerault we just render a list of buffers we get from neovim, they don't come with metadata regarding the last time they were access, not sure if they have a modified timestamp (but there is a modified boolean), in order to get the MRU I think its possible to query neovim via the api for the most recently used files as I believe theres a vim command for that (`:oldfiles`) possibly the output would give you something like that but it gives all the files aka in different projects as well but I guess those could be filtered out based on the current working dir.

All in all I think it would be great to have but would essentially be a new feature to add to the plugin that would require a bit of legwork. It might possibly be as straight forward as

```ts
	const getMRU = await Oni.neovim.executeCommand("oldfiles")
	// Hopefully we get an array of files ["1: blah.txt", ...]
	// then we could filter those and render that instead
```
Making a *bunch* of assumptions though

**Comment by parkerault** [comment]
You're right, I guess those buffer plugins are doing some heavy monitoring behind the scenes to get the MRU sorting. I didn't see anything in the documentation that suggests that vim can return different sort orders natively. Since this seems like a non-trivial feature I'm fine with closing this issue, unless someone else really wants it.

**Comment by Akin909** [comment]
I think its likely to come up again I've definitely thought it'd be cool to have might be worth having as a discussion point in case someone has a better suggestion re hot to implement it.

One thought I had would be finding out how `fzf` does it which I think is in vimscript querying the info from vim and somehow rendering it so we could possibly borrow from that implementation

**Comment by parkerault** [comment]
Buffergator creates a MRU array and uses autocommands on BufEnter, BufRead, BufNewFile, and BufWritePost to update the array. It looks pretty complicated and definitely doesn't look like vim gives them any help: https://github.com/jeetsukumaran/vim-buffergator/blob/master/plugin/buffergator.vim
